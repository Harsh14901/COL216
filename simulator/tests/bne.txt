main:
  addi $t0, $zero, 0
  addi $s0, $zero, 5
start:
  beq $t0, $s0, exit
  addi $t0, $t0, 1
  j start
exit: