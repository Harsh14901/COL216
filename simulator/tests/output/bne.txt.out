[+] Compiling file : tests/input/bne.txt
[+] Program compiled successfully !
[+] Executing program ...
[+] Executing in NON-BLOCKING MODE
---------- Execution logs ----------
[$] CYCLE 1-1 :
[#] Executing current instruction -> addi $t0, $zero, 0
[#] Register contents -
	 $zero :          0		   $s0 :          0
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          0		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 2-2 :
[#] Executing current instruction -> addi $s0, $zero, 5
[#] Register contents -
	 $zero :          0		   $s0 :          5
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          0		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 3-3 :
[#] Executing current instruction -> beq $t0, $s0, exit
[#] Register contents -
	 $zero :          0		   $s0 :          5
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          0		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 4-4 :
[#] Executing current instruction -> addi $t0, $t0, 1
[#] Register contents -
	 $zero :          0		   $s0 :          5
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          1		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 5-5 :
[#] Executing current instruction -> j start
[#] Register contents -
	 $zero :          0		   $s0 :          5
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          1		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 6-6 :
[#] Executing current instruction -> beq $t0, $s0, exit
[#] Register contents -
	 $zero :          0		   $s0 :          5
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          1		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 7-7 :
[#] Executing current instruction -> addi $t0, $t0, 1
[#] Register contents -
	 $zero :          0		   $s0 :          5
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          2		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 8-8 :
[#] Executing current instruction -> j start
[#] Register contents -
	 $zero :          0		   $s0 :          5
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          2		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 9-9 :
[#] Executing current instruction -> beq $t0, $s0, exit
[#] Register contents -
	 $zero :          0		   $s0 :          5
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          2		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 10-10 :
[#] Executing current instruction -> addi $t0, $t0, 1
[#] Register contents -
	 $zero :          0		   $s0 :          5
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          3		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 11-11 :
[#] Executing current instruction -> j start
[#] Register contents -
	 $zero :          0		   $s0 :          5
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          3		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 12-12 :
[#] Executing current instruction -> beq $t0, $s0, exit
[#] Register contents -
	 $zero :          0		   $s0 :          5
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          3		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 13-13 :
[#] Executing current instruction -> addi $t0, $t0, 1
[#] Register contents -
	 $zero :          0		   $s0 :          5
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          4		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 14-14 :
[#] Executing current instruction -> j start
[#] Register contents -
	 $zero :          0		   $s0 :          5
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          4		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 15-15 :
[#] Executing current instruction -> beq $t0, $s0, exit
[#] Register contents -
	 $zero :          0		   $s0 :          5
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          4		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 16-16 :
[#] Executing current instruction -> addi $t0, $t0, 1
[#] Register contents -
	 $zero :          0		   $s0 :          5
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          5		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 17-17 :
[#] Executing current instruction -> j start
[#] Register contents -
	 $zero :          0		   $s0 :          5
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          5		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 18-18 :
[#] Executing current instruction -> beq $t0, $s0, exit
[#] Register contents -
	 $zero :          0		   $s0 :          5
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          5		   $t8 :          0
	   $t1 :          0		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 19-19 :
[#] Executing current instruction -> addi $t1, $zero, 1
[#] Register contents -
	 $zero :          0		   $s0 :          5
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          5		   $t8 :          0
	   $t1 :          1		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
[$] CYCLE 20-20 :
[#] Executing current instruction -> addi $t1, $zero, 1
[#] Register contents -
	 $zero :          0		   $s0 :          5
	   $at :          0		   $s1 :          0
	   $v0 :          0		   $s2 :          0
	   $v1 :          0		   $s3 :          0
	   $a0 :          0		   $s4 :          0
	   $a1 :          0		   $s5 :          0
	   $a2 :          0		   $s6 :          0
	   $a3 :          0		   $s7 :          0
	   $t0 :          5		   $t8 :          0
	   $t1 :          1		   $t9 :          0
	   $t2 :          0		   $k0 :          0
	   $t3 :          0		   $k1 :          0
	   $t4 :          0		   $gp :          0
	   $t5 :          0		   $sp :    1048576
	   $t6 :          0		   $fp :          0
	   $t7 :          0		   $ra :          0
[#] Row buffer updates:
[#] Memory updates:
[#] Remarks: 
============================================================
---------- Execution statistics ----------
[#] Clock cycles: 20
[#] Number of row buffer updates: 0
[#] Updated memory : 
[+] Program terminated
